backend/api/CourseApi/getStudentsStats.js

```diff
@@ -0,0 +1,54 @@
+import knex from '~/db/knex';
+import auth from '~/middlewares/auth';
+
+const getStudentsStats = auth(async (request, response) => {
+
+  const authorId = request.body['authorId'];
+  const courseId = request.body['courseId'];
+
+  const students = await knex('user')
+    .select('user.*')
+    .join('course_user_is_learning', { 'course_user_is_learning.user_id': 'user.id' })
+    .where({ 'course_user_is_learning.courseId': courseId, active: true });
+
+  const totalAmountOfCards = await knex('problem').select().where({ courseId });
+  const promises = []
+
+  students.forEach((student) => {
+    
+    const stats = knex('problem_user_is_learning')
+      .select()
+      .whereIn('course_user_is_learning_id', function() {
+        this.select('id').from('course_user_is_learning').where({
+          user_id: student.id,
+          course_id: courseId
+        })
+    })
+
+    promises.push(stats)
+  });
+
+  const filteredProblems = await Promise.all(promises);
+
+  const dto = []
+
+  for (let index = 0; index < filteredProblems.length; index++) {
+    const problems = filteredProblems[index];
+    const latestReviewedFlashcard = problems.sort((a, b) => b.problemId - a.problemId)[0]
+
+    dto.push({
+      id: students[index].id,
+      username: students[index].username,
+      avatarUrl: students[index].avatarUrl,
+      lastReviewedAt: latestReviewedFlashcard.lastReviewedAt,
```

Comment:
Ah, there's an error in the terminal:

<img width="642" alt="image" src="https://user-images.githubusercontent.com/7578559/155733276-8c02e159-5bfc-4da4-a173-9c785332dd71.png">

1. There exist flashcards that were learned, but not yet reviewed - meaning their `lastReviewedAt` is null.
2. There exist students who haven't yet learned any flashcard, so `problems` is of length 0, and `latestReviewedFlashcard` is undefined.

Reply:
Ok, I think I added a fix for this with my latest commit:

![last_reviewed_at](https://user-images.githubusercontent.com/98980760/155811645-81ca00e5-dd3b-4cf1-a90e-6d926db12835.png)